src/precompiles/tdx_dcap.rs

```diff
@@ -0,0 +1,160 @@
+use revm::precompile::{
+    Precompile, PrecompileError, PrecompileResult, PrecompileWithAddress, StandardPrecompileFn,
+};
+
+use std::ffi;
+
+use ethers::abi::{encode_packed, Detokenize, Function, Token};
+use ethers::types::H160;
+use ethers::utils::keccak256;
+use sha2::*;
+
+use lazy_static::lazy_static;
+use std::{collections::HashMap, sync::Mutex};
+use std::{fs, fs::File, io::Read, io::Write, path::Path};
+
+use crate::{u64_to_address, QuoteVerificationLibrary};
+
+use ethers::abi::{encode, Bytes as AbiBytes, Contract};
+use ethers::contract::{BaseContract, Lazy};
+use ethers::types::{Address, Bytes, H256};
+
+use revm::primitives::{Address as RevmAddress, Env};
+
+pub const VERIFY_QUOTE: PrecompileWithAddress = PrecompileWithAddress::new(
+    u64_to_address(0x40800),
+    Precompile::Standard(tgx_verify_quote_run as StandardPrecompileFn),
+);
+
+// Redefined here to avoid having to import from external services
+pub static TDX_DCAP_ABI: Lazy<BaseContract> = Lazy::new(|| {
+    let contract: Contract =
+        serde_json::from_str(include_str!("../out/TdxDcap.sol/TDX_DCAP.abi.json")).unwrap();
+    BaseContract::from(contract)
+});
+
+pub static VERIFY_QUOTE_ABI: Lazy<Function> = Lazy::new(|| {
+    TDX_DCAP_ABI
+        .abi()
+        .function("verifyQuote")
+        .expect("verifyQuote signature not available in TDX DCAP abi")
+        .clone()
+});
+
+const INPUTS_TOO_BIG: PrecompileError =
+    PrecompileError::CustomPrecompileError("inputs passed in are unreasonably big");
+const INCORRECT_INPUTS: PrecompileError =
+    PrecompileError::CustomPrecompileError("incorrect inputs passed in");
```

Comment:
nit would use alloy for this https://docs.rs/alloy-core/latest/alloy_core/sol_types/macro.sol.html, 

https://github.com/alloy-rs/examples/blob/5b6ceb628a027eeb33feef1135ed10fcf0a7468a/examples/sol-macro/examples/decode_returns.rs

https://github.com/alloy-rs/examples/blob/5b6ceb628a027eeb33feef1135ed10fcf0a7468a/examples/transactions/examples/decode_input.rs#L23-L44